controller_test.go

```diff
@@ -264,6 +264,18 @@ func (s *controllerSuite) TestMachinesArgs(c *gc.C) {
 	c.Assert(request.URL.Query(), gc.HasLen, 6)
 }
 
+func (s *controllerSuite) TestMachinesArgsValidate(c *gc.C) {
+	args := MachinesArgs{
+		Hostnames: []string{"helpful-bear"},
+		Tags: map[string]string{
+			"juju-model":         "aaa-bbb",
+			"juju-is-controller": "true",
+		},
+	}
+	err := args.Validate()
+	c.Assert(err, gc.ErrorMatches, "searching by tags with other filters not supported")
```

Comment:
Sorry, there was a bit of churn in this change (because I didn't know about set-owner-data I was faking it horribly using tags) - this change is removed in a subsequent commit.